src/IceRpc/Transports/Slic/Internal/SlicConnection.cs

```diff
@@ -265,9 +264,7 @@ async Task<TransportConnectionInformation> PerformConnectAsync()
 
             if (idleTimeout != Timeout.InfiniteTimeSpan)
             {
-                // Only client connections send ping frames when idle to keep the server connection alive. The server
-                // sends back a Pong frame in turn to keep alive the client connection.
-                _enableIdleTimeoutAndKeepAlive(idleTimeout, IsServer ? null : KeepAlive);
+                _duplexConnection.Enable(idleTimeout);
```

Comment:
Would be better to keep `IDuplexConnection` and then do 

```
if (_duplexConnection is SlicDuplexConnectionDecorator duplexConnectionDecoator)
{
    duplexConnectionDecoator.Enable();
} 
```

Then we can make `SlicDuplexConnectionDecorator` timers non-nullable, and keep a single constructor.

Reply:
I don't see how this would work.

The difficulty here is the idle timeout is negotiated during connection establishment. At construction time, we don't know if the negotiated idle timeout is good to be infinite or some other value.

Reply:
We can pass the `idleTimeout` I forget to add it there, and that is not the point. The point is to avoid creating this decorator for the Server connections.

Reply:
We need the decorator for server connections too. It implements the idle timer.